File: BankKata/BankKataTests/UnitTests/ClockShould.cs
using BankCore;
using NUnit.Framework;
using System;
using BankCore.Models;

namespace BankKataTests.UnitTests
{
    [TestFixture]
    public class ClockShould
    {
        private const string TODAY = "07/04/2016";

        [Test]
        public void Return_Todays_Date_In_Dd_Mm_Yyyy_Format()
        {
            var clock = new TestableClock();

            Assert.That(clock.TodayAsString, Is.EqualTo(TODAY));
        }

        private class TestableClock : Clock
        {
            protected override string Today()
            {
                return "07/04/2016";
            }
        }
    }
}

File: BankKata/BankKataTestsVBNetEdition/UnitTests/ClockShould.vb
Imports BankCoreVBNetEdition
Imports NUnit.Framework

Namespace UnitTests

    <TestFixture()>
    Public Class ClockShould

        Protected Const TestDate As String = "07/04/2016"

        <Test()>
        Public Sub Return_Todays_Date_In_Dd_Mm_Yyyy_Format()
            Dim clock As New TestableClock()

            Assert.That(clock.TodayAsString(), [Is].EqualTo(TestDate))
        End Sub

        Public Class TestableClock
            Inherits Clock

            Protected Overrides Function Today() As String
                Return TestDate
            End Function
        End Class
    End Class
End NameSpace